Wikipedia/UI-V5/WMFArticleContainerViewController.m

```diff
@@ -452,6 +459,27 @@ - (CGFloat)totalProgressWithArticleFetcherProgress:(CGFloat)progress {
     return 0.8 * progress;
 }
 
+#pragma mark - Significantly Viewed Timer
+
+- (void)startSignificantlyViewedTimer {
+    if(self.significantlyViewedTimer){
+        return;
+    }
+    MWKHistoryList* historyList = self.dataStore.userDataStore.historyList;
+    MWKHistoryEntry* entry      = [historyList entryForTitle:self.articleTitle];
+    if (!entry.titleWasSignificantlyViewed) {
+        self.significantlyViewedTimer = [NSTimer scheduledTimerWithTimeInterval:FBTweakValue(@"Home", @"Related items", @"Required viewing time", 30.0) target:self selector:@selector(significantlyViewedTimerFired:) userInfo:nil repeats:NO];
```

Comment:
missing cancel/resume on `viewDidDisappear`/`viewDidAppear`, which means this will always be marked as significant since `NSTimer` retains its target—which also means our VCs are living for however long the timeout is.